src/pages/SupplierRequisitionsPage.js

```diff
@@ -1,271 +1,198 @@
-        defaultSortDirection={this.dataFilters.isAscending ? 'ascending' : 'descending'}
-        columns={[
-          {
-            key: 'serialNumber',
-            width: 1.5,
-            title: tableStrings.requisition_number,
-            sortable: true,
-          },
-          {
-            key: 'supplierName',
-            width: 2,
-            title: tableStrings.supplier,
-            sortable: false,
-          },
-          {
-            key: 'entryDate',
-            width: 1,
-            title: tableStrings.entered_date,
-            sortable: true,
-          },
-          {
-            key: 'numberOfItems',
-            width: 1,
-            title: tableStrings.items,
-            sortable: true,
-            alignText: 'right',
-          },
-          {
-            key: 'status',
-            width: 1,
-            title: tableStrings.status,
-            sortable: true,
-          },
-          {
-            key: 'delete',
-            width: 1,
-            title: tableStrings.delete,
-            alignText: 'center',
-          },
-        ]}
-        dataTypesSynchronised={DATA_TYPES_SYNCHRONISED}
-        database={database}
-        selection={selection}
-        {...genericTablePageStyles}
-        topRoute={topRoute}
-      >
-        <BottomConfirmModal
-          isOpen={selection.length > 0}
-          questionText={modalStrings.delete_these_invoices}
-          onCancel={this.onDelete(false)}
-          onConfirm={this.onDelete(true)}
-          confirmText={modalStrings.delete}
-        />
-
-        <SelectModal
-          isOpen={isCreatingRequisition}
-          options={database.objects('InternalSupplier')}
-          placeholderText={modalStrings.start_typing_to_select_supplier}
-          queryString="name BEGINSWITH[c] $0"
-          sortByString="name"
-          onSelect={name => this.onNewRequisition({ otherStoreName: name })}
-          onClose={() => this.onNewRequisition()}
-          title={modalStrings.search_for_the_supplier}
-        />
-
-        {byProgramModalOpen && (
-          <ByProgramModal
-            isOpen={byProgramModalOpen}
-            onConfirm={this.onNewRequisition}
-            onCancel={this.onNewRequisition}
-            database={database}
-            transactionType="requisition"
-            settings={settings}
+  const renderHeader = () => (
+    <DataTableHeaderRow
+      columns={columns}
+      dispatch={instantDebouncedDispatch}
+      sortAction={sortData}
+      isAscending={isAscending}
+      sortBy={sortBy}
+    />
+  );
+
+  const {
+    newPageTopSectionContainer,
+    newPageTopLeftSectionContainer,
+    newPageTopRightSectionContainer,
+    searchBar,
+  } = newPageStyles;
+  return (
+    <DataTablePageView>
+      <View style={newPageTopSectionContainer}>
+        <View style={newPageTopLeftSectionContainer}>
+          <SearchBar
+            onChange={value => debouncedDispatch(filterData(value))}
```

Comment:
```suggestion
            onChange={handleSearch}
```

with

```
const handleSearch value => debouncedDispatch(filterData(value));
```

declared somewhere outside of the `render()` method.
Would prevent creation of a method within render.